modules/mbedtls/Kconfig.tls-generic

```diff
@@ -327,6 +327,10 @@ config MBEDTLS_MAC_SHA256_ENABLED
 	bool "SHA-224 and SHA-256 hash algorithms"
 	default y
 
+config PSA_WANT_ALG_SHA_256
```

Comment:
Excellent that we are using PSA_WANT_ALG_SHA256 but I wonder if it is possible to right now remove the dependency on BUILT_WITH_TFM as a hash-algorithm doesn't require or assume physical storage and/or key management support...

As long as it is handled in `config-tls-generic.h` and as long as we have a signal e.g. to state that we want to build the PSA core (providing PSA crypto API support) locally when BUILD_WITH_TFM is not set, then I see no issue with removing this dependency.

Hopefully just a small addition to the existing PR (have you tried just removing this depends?)

Reply:
Right, `BUILD_WITH_TFM` is too strict as requirement since the PSA crypto might be builtin. However I'm not sure we can always remove the legacy path. What if the user does not enable TFM and he/she also keeps `CRYPTO_C` disabled in MbedTLS? Then there would be no PSA crypto provider for the hash computation... Am I wrong?

Reply:
At least the `depends on` could be extended to require either TF-M or the PSA core, if there is an appropriate Kconfig option that defines the latter. This would give (not sure about the Kconfig symbol): `depends on BUILD_WITH_TFM || MBEDTLS_PSA_CRYPTO_C`

Reply:
> At least the `depends on` could be extended to require either TF-M or the PSA core, if there is an appropriate Kconfig option that defines the latter. This would give (not sure about the Kconfig symbol): `depends on BUILD_WITH_TFM || MBEDTLS_PSA_CRYPTO_C`

Unfortunately, as discussed on Teams, this is not possible because because `MBEDTLS_PSA_CRYPTO_C` depends on `MBEDTLS_SHA256_C` so any attempt to make `PSA_WANT_ALG_SHA_256` enabled when the legacy is not present creates a depedency loop in Kconfig options.

> Hopefully just a small addition to the existing PR (have you tried just removing this depends?)

As requested on Teams I opened a parallel PR for this: #71827. It is functionally identical to this one a part from the fact that it has no `BUILD_WITH_TFM` and it  only relies on PSA functions (no legacy alternative).